src/frontend/src/optimizer/plan_node/stream_sink.rs

```diff
@@ -487,7 +487,8 @@ impl StreamSink {
         ) {
             (true, true, _) => Ok(SinkType::AppendOnly),
             (false, true, true) => Ok(SinkType::ForceAppendOnly),
-            (_, false, false) => Ok(SinkType::Upsert),
+            (true, false, false) => Ok(SinkType::AppendOnly),
```

Comment:
FYI this also covers `format upsert` (in addition to omitting `type = `). So when the upstream is append-only and user explicitly writes `format upsert`, the resulting sink will also be `append-only`.